packages/browser-wallet-api/test/compatibility.test.ts

```diff
@@ -501,7 +503,33 @@ describe(sanitizeSendTransactionInput, () => {
         };
         const result = sanitizeSendTransactionInput(accountAddress, type, payload);
         expect(result).toEqual(expected);
-        expect(result.payload).toBe(payload);
+        expect((result.payload as RegisterDataPayload).data).toStrictEqual(payload.data);
+    });
+
+    test('Transformed "RegisterData" transaction input with "DataBlob"-like parameter can be parsed', () => {
+        const type = AccountTransactionType.RegisterData;
+
+        const payload: RegisterDataPayload = {
+            data: {
+                __type: 'ccd_block_hash',
+                data: new DataBlob(Buffer.from('This is data!')).data,
+            } as unknown as DataBlob,
+        };
+
+        const expected: SanitizedSendTransactionInput = {
+            accountAddress: AccountAddress.fromBase58(accountAddress),
+            type,
+            payload: {
+                data: new DataBlob(Buffer.from('This is data!')),
+            },
+        };
+
+        const result = sanitizeSendTransactionInput(accountAddress, type, payload);
+
+        const parsed = jsonParse(jsonStringify(result));
```

Comment:
Isn't this overkill for this test? Just doing the sanitize should do the trick?

Reply:
No, because just checking `expect(result).toEqual(expected);`
 doesn't catch the error that I ran into. :upside_down_face: 
 (Also that version of the test succeeds on main)